### DISCUSSION TITLE
How to use to transcribe specific audio files

### DISCUSSION THREAD

**Discussion by Soeroah** [openai/whisper#2387] [Q&A] [answered]
Hi,

I'm pretty unskilled at python so I usually need help with command lines.

I've got some short, 1-4 minute files I'd like to get transcribed and this project was recommended. However, each time I run the command I get a progress bar showing 1.5 gigabytes of data being processed, when the total size of the files in the folder I want transcribed are sub-100 megabytes. 

I'm pretty sure the program is just grabbing audio files from somewhere on my PC but I have no idea where. How do I customise which files specifically I want to have transcribed?

Thank you.

**Comment by glangford** [comment, marked as answer]
You can change to the directory of your audio files using the `cd` command. Then run whisper. Or you can specify the path to your files, as suggested here

* https://github.com/openai/whisper/discussions/2091#discussioncomment-8815928

If you have a number of files of the same type (eg. .wav or .mp3) you can just list them one by one on the command line, or you can process all the files of one type like this 

`whisper *.mp3 --language en --verbose True --output_format txt`

To see the different options, use

`whisper --help`

**Reply by Soeroah** [reply]
Hm, I tried this but I'm getting the same error. However, I think the verbose command gave me extra information?

"UserWarning: C:\Users\David\.cache\whisper\large-v3-turbo.pt exists, but the SHA256 checksum does not match; re-downloading the file
  warnings.warn(
  5%|█▋                                   | 71.7M/1.51G [00:33<08:07, 3.16MiB/s]"

When I originally downloaded whisper it was in the realm of a few hundred megabytes, is this size normal for a redownload?

**Reply by glangford** [reply]
This warning is saying that whisper is downloading the *model* used to transcribe the speech. 

See these past discussions to try and troubleshoot the checksum error
* https://github.com/openai/whisper/discussions/1027
* https://github.com/openai/whisper/discussions/1218

**Reply by Soeroah** [reply]
Thank you. I let it run to completion and it seems to be working, though it takes a little while for transcription to start.

I think I've got the hang of it now

**Reply by Soeroah** [reply]
Actually, sorry to reopen - I've tried running the "do all files of this type" as you suggested above but I get the following error

> C:\Users\David\Documents\SoundRecordMp3\SoundRecordMp3\bin\Debug\Output>
> C:\Users\David\Documents\SoundRecordMp3\SoundRecordMp3\bin\Debug\Output>whisper *.mp3 --language en --verbose True --output_format txt --output_dir "C:\Users\David\Documents\SoundRecordMp3\SoundRecordMp3\bin\Debug\Output"
> C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\__init__.py:150: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
>   checkpoint = torch.load(fp, map_location=device)
> C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\transcribe.py:126: UserWarning: FP16 is not supported on CPU; using FP32 instead
>   warnings.warn("FP16 is not supported on CPU; using FP32 instead")
> Traceback (most recent call last):
>   File "C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\audio.py", line 58, in load_audio
>     out = run(cmd, capture_output=True, check=True).stdout
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 571, in run
>     raise CalledProcessError(retcode, process.args,
> subprocess.CalledProcessError: Command '['ffmpeg', '-nostdin', '-threads', '0', '-i', '*.mp3', '-f', 's16le', '-ac', '1', '-acodec', 'pcm_s16le', '-ar', '16000', '-']' returned non-zero exit status 4294967274.
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\transcribe.py", line 597, in cli
>     result = transcribe(model, audio_path, temperature=temperature, **args)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\transcribe.py", line 133, in transcribe
>     mel = log_mel_spectrogram(audio, model.dims.n_mels, padding=N_SAMPLES)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\audio.py", line 140, in log_mel_spectrogram
>     audio = load_audio(audio)
>             ^^^^^^^^^^^^^^^^^
>   File "C:\Users\David\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\audio.py", line 60, in load_audio
>     raise RuntimeError(f"Failed to load audio: {e.stderr.decode()}") from e
> RuntimeError: Failed to load audio: ffmpeg version 2024-07-24-git-896c22ef00-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
>   built with gcc 13.2.0 (Rev5, Built by MSYS2 project)
>   configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libopenjpeg --enable-libquirc --enable-libuavs3d --enable-libxevd --enable-libzvbi --enable-libqrencode --enable-librav1e --enable-libsvtav1 --enable-libvvenc --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxeve --enable-libxvid --enable-libaom --enable-libjxl --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-dxva2 --enable-d3d11va --enable-d3d12va --enable-ffnvcodec --enable-libvpl --enable-nvdec --enable-nvenc --enable-vaapi --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
>   libavutil      59. 28.100 / 59. 28.100
>   libavcodec     61. 10.100 / 61. 10.100
>   libavformat    61.  5.101 / 61.  5.101
>   libavdevice    61.  2.100 / 61.  2.100
>   libavfilter    10.  2.102 / 10.  2.102
>   libswscale      8.  2.100 /  8.  2.100
>   libswresample   5.  2.100 /  5.  2.100
>   libpostproc    58.  2.100 / 58.  2.100
> [in#0 @ 00000213e1a72ac0] Error opening input: Invalid argument
> Error opening input file *.mp3.
> Error opening input files: Invalid argument
> 
> Skipping *.mp3 due to RuntimeError: Failed to load audio: ffmpeg version 2024-07-24-git-896c22ef00-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
>   built with gcc 13.2.0 (Rev5, Built by MSYS2 project)
>   configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libopenjpeg --enable-libquirc --enable-libuavs3d --enable-libxevd --enable-libzvbi --enable-libqrencode --enable-librav1e --enable-libsvtav1 --enable-libvvenc --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxeve --enable-libxvid --enable-libaom --enable-libjxl --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-dxva2 --enable-d3d11va --enable-d3d12va --enable-ffnvcodec --enable-libvpl --enable-nvdec --enable-nvenc --enable-vaapi --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
>   libavutil      59. 28.100 / 59. 28.100
>   libavcodec     61. 10.100 / 61. 10.100
>   libavformat    61.  5.101 / 61.  5.101
>   libavdevice    61.  2.100 / 61.  2.100
>   libavfilter    10.  2.102 / 10.  2.102
>   libswscale      8.  2.100 /  8.  2.100
>   libswresample   5.  2.100 /  5.  2.100
>   libpostproc    58.  2.100 / 58.  2.100
> [in#0 @ 00000213e1a72ac0] Error opening input: Invalid argument
> Error opening input file *.mp3.
> Error opening input files: Invalid argument

I think if I do it one by one with a full filepath in the line it works, but for some reason doing all at once doesn't?

**Reply by glangford** [reply]
Ah...I see you are on Windows, and it doesn't like '*' to wildcard, which is the standard way to do this on other platforms. My mistake.

> Error opening input file *.mp3.
Error opening input files: Invalid argument

See here for more information

Is there any way to get the windows cmd shell to expand wildcard paths
* https://superuser.com/questions/460598/is-there-any-way-to-get-the-windows-cmd-shell-to-expand-wildcard-paths?rq=1

**Reply by Soeroah** [reply]
Ah, damn, powershell.. I've got no idea how to modify the whisper command to work with that

I'll look into it but if anyone happens to know the answer I'll be looking for I'd appreciate the help. Like I said, I'm a pretty basic non-programmer user

Edit: For anyone in the future looking, I found one that works for batching with windows called Faster-Whisper-XXL
